propulsion-hotel/Reactor/SourceArgs.fs

```diff
@@ -0,0 +1,119 @@
+        | [<AltCommandLine "-r"; Unique>]   ReadConnectionString of string
+        | [<AltCommandLine "-cp">]          CheckpointConnectionString of string
+        | [<AltCommandLine "-s">]           Schema of string
+        | [<AltCommandLine "-b"; Unique>]   BatchSize of int
+        | [<AltCommandLine "-Z"; Unique>]   FromTail
+        interface IArgParserTemplate with
+            member a.Usage = a |> function
+                | ConnectionString _ ->     $"Connection string for the postgres database housing message-db when writing. (Optional if environment variable {CONNECTION_STRING} is defined)"
+                | ReadConnectionString _ -> $"Connection string for the postgres database housing message-db when reading. (Defaults to the (write) Connection String; Optional if environment variable {READ_CONN_STRING} is defined)"
+                | CheckpointConnectionString _ -> "Connection string used for the checkpoint store. If not specified, defaults to the connection string argument"
+                | Schema _ ->               $"Schema that should contain the checkpoints table Optional if environment variable {SCHEMA} is defined"
+                | BatchSize _ ->            "maximum events to load in a batch. Default: 100"
+                | FromTail _ ->             "(iff the Consumer Name is fresh) - force skip to present Position. Default: Never skip an event."
+
+    type Arguments(c : Args.Configuration, p : ParseResults<Parameters>) =
+        let writeConnStr =                  p.TryGetResult ConnectionString |> Option.defaultWith (fun () -> c.MdbConnectionString)
+        let readConnStr =                   p.TryGetResult ReadConnectionString |> Option.orElseWith (fun () -> c.MdbReadConnectionString) |> Option.defaultValue writeConnStr
+        let checkpointConnStr =             p.TryGetResult CheckpointConnectionString |> Option.defaultValue writeConnStr
+        let schema =                        p.TryGetResult Schema |> Option.defaultWith (fun () -> c.MdbSchema)
+        let fromTail =                      p.Contains FromTail
+        let batchSize =                     p.GetResult(BatchSize, 100)
+        let tailSleepInterval =             TimeSpan.FromMilliseconds 500.
+        member _.Connect() =
+                                            let connStrWithoutPassword = Npgsql.NpgsqlConnectionStringBuilder(checkpointConnStr, Password = null)
+                                            Log.Information("MessageDb connection {connectionString}", connStrWithoutPassword.ToString())
+                                            let client = Equinox.MessageDb.MessageDbClient(writeConnStr, readConnStr)
+                                            Equinox.MessageDb.MessageDbContext(client, batchSize)
+        member _.MonitoringParams(log : ILogger) =
+            log.Information("MessageDbSource batchSize {batchSize} Checkpoints schema {schema}", batchSize, schema)
```

Comment:
Not sure what your standard is for application of `=`. I've seen it a lot, but also `: `

```suggestion
            log.Information("MessageDbSource batchSize={batchSize} Checkpoints schema={schema}", batchSize, schema)
```

Reply:
In Serilog this is de-emphasized in docs and samples
- you've done something wrong if you're having to regex (LCD should be json rendering as opposed to text, which can be queried)
- on console, its syntax highlighted

I do know it is sometimes neat and a pretty well established convention
But, so is *not* having it in all of github.com/jet !